ql/src/java/org/apache/hadoop/hive/ql/ddl/function/AbstractFunctionAnalyzer.java

```diff
@@ -70,7 +71,7 @@ protected void addEntities(String functionName, String className, boolean isTemp
       }
     }
     if (database != null) {
-      outputs.add(new WriteEntity(database, WriteEntity.WriteType.DDL_NO_LOCK));
+      inputs.add(new ReadEntity(database));
```

Comment:
You should only do this if the query is drop function. That's the reason why we see the difference in .q.out changes with create function as well.

Reply:
Don't we need this change for Create Function? I and Naveen were discussing this change the other day and thought that Read privilege on DB should be good for Create function too. Please share your thoughts if you think otherwise.

Reply:
Yeah, for create/drop we only need 'select/read' privileges.